### ISSUE TITLE
Failed to load Doom: Profile init file hasn’t been generated. Did you forgot to run ’doom sync’?

### ISSUE THREAD

**Issue by ryukinix** [doomemacs/doomemacs#6926]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [X] The issue can be reproduced on the **latest** available commit of Doom.

- [X] The issue can be reproduced on a stable release of Emacs, such as 27 or 28. *(Doom does not support development builds like 29+ or any version ending in .50 or .9x)*

### Expected behavior

No issues, no flycheck errors.

### Current behavior

I cloned my doomemacs as `~/.emacs.d.doomemacs` to don't mixup with my main config.

When I try to edit any file from `~/.doom.d/` I got the flycheck error:

```
Failed to load Doom: Profile init file hasn’t been generated. Did you forgot to run ’doom sync’?
```

![image](https://user-images.githubusercontent.com/7642878/198887663-937e329c-5ff3-4e68-b2a7-9e5f2e9d953f.png)

### Steps to reproduce

1. Install doomemacs in a non-default path (ex.: `~/.emacs.d.doomemacs`)
2. Run `doom sync`
3. Open any `~/.doom.d/` file, for instance config.el

### System Information

https://gist.github.com/ryukinix/d5e47226df1cd256c780b07c532b24f7

**Comment by hlissner** [comment]
Try `doom sync -p` (this will purge orphaned packages and leftover *.eln files).

**Comment by ryukinix** [comment]
> Try doom sync -p (this will purge orphaned packages and leftover *.eln files).

Thank you for your suggestion, this fixed another problem I had in my editor. However, the problem related remains. It only works if I use `~/.emacs.d` path.

**Comment by hlissner** [comment]
Oh, I misread. How are you launching Doom? With `doom run`? Are you using [doom profiles](https://github.com/doomemacs/doomemacs/tree/master/profiles)?

**Comment by ryukinix** [comment]
> Oh, I misread. How are you launching Doom? With doom run? Are you using [doom profiles](https://github.com/doomemacs/doomemacs/tree/master/profiles)?

Yes, i'm running with `doom run`. I tried setup doomemacs profiles, but I think didn't understand that properly yet.

> However, for this to work, Doom must live in =~/.emacs.d= or =~/.config/emacs=.

Since I'm storing doom repository at `~/.emacs.d.doomemacs`, that seems not be supported at all.

**Comment by jackjliew** [comment]
I have the same issue when using chemacs to switch Doom and Spacemacs. No other problems found but only the Flycheck error is a bit annoying.

**Comment by jasonjckn** [comment]
I have the same issue when using chemacs

**Comment by guillaume-d** [comment]
Same here without neither Chemacs nor doom profiles, `doom sync -p` did not help.
I installed using ``~/.local/share/doom/bin/doom --doomdir ~/.config/doom install`` and starts using ``~/.local/share/doom/bin/doom run``
``--doomdir`` seems superfluous for all `doom` subcommands after being set once on `doom install`, however I also tried with it and I made no difference.

Here strangely the message only appears when editing ``~/.config/doom/config.el``, but not other Emacs LISP files or a Markdown one there, and other directories seem also fine.

I do not mind the occasional glitch, but as a newbie Doom Emacs user and as a (Emacs) LISP amateur, I wonder whether my setup is correct and whether my configuration is applied completely.

**Comment by weidaolee** [comment]
> I have the same issue when using chemacs to switch Doom and Spacemacs. No other problems found but only the Flycheck error is a bit annoying.

same

**Comment by A7R7** [comment]
I have the same issue when launching doom emacs from `emacs --init-directory=~/.doom.d`, so this is may not be the problem of chemacs.

**Comment by sahlte** [comment]
Here is some info I found while digging around:

If you search the doom codebase for this error, you will find only one place where it could be thrown, it is in this code block in lisp/doom-start.el:

```
;;; Load the rest of $DOOMDIR + modules if noninteractive
;; If the user is loading this file from a batch script, let's assume they want
;; to load their userland config as well.
(when noninteractive
  (doom-require 'doom-profiles)
  (let ((init-file (doom-profile-init-file)))
    (unless (file-exists-p init-file)
      (user-error "Profile init file hasn't been generated. Did you forgot to run 'doom sync'?"))
    (let (kill-emacs-query-functions
          kill-emacs-hook)
      ;; Loads modules, then $DOOMDIR/config.el
      (doom-load init-file 'noerror)
      (doom-initialize-packages))))
```

What I find strange about this is the fact that this code should only be running if emacs is loaded noninteractively, however I am starting emacs to use the gui. 

My only real thought about this is that something is going wrong due to me using chemacs2, perhaps when I run "emacs --with-profile doom" to start doom, chemacs is running it noninteractively at first. However, I looked into it and this doesn't seem to be the case.

The other strange thing is that when I eval this test code in a buffer:
```
(doom-require 'doom-profiles)
(doom-profile-init-file)
```
The return value of the second expression is:

```
"<MY HOME PATH aka ~>/emacs_profiles/doomacs/.local/etc/@/init.29.el"
```
This is strange because seemingly the error is only supposed to be printed if this function returned nil, but as you can see it isn't nil when I run it. And yes, this file does in fact exist.

Another strange behavior is that if I add a print statement within the unless statement, I never see that printed to the messages buffer, even if I open an my .doom.d/config.el file. However, I know that the error is being thrown by this piece of code because it doesn't happen when it is removed (with no harm seemingly?)
